generative_ai/embedding_preview.py

```diff
@@ -18,13 +18,31 @@
 from vertexai.language_models import TextEmbeddingInput, TextEmbeddingModel
 
 
-def embed_text(
-    texts: List[str] = ["banana muffins? ", "banana bread? banana muffins?"],
-    task: str = "QUESTION_ANSWERING",
-    model_name: str = "text-embedding-preview-0409",
+def embed_code_query(
+    texts: List[str] = ["Retrieve a function that adds two numbers"],
+    task: str = "CODE_RETRIEVAL_QUERY",
+    model_name: str = "text-embedding-preview-0815",
     dimensionality: Optional[int] = 256,
 ) -> List[List[float]]:
```

Comment:
As per [PEP-585](https://peps.python.org/pep-0585/), we can use `list` directly, so this can be changed to `list[list[float]]`, and the typing import can be removed.

Reply:
Since we're passing the python 3.8 build, I think we may have to use List instead of list. WDYT?

Reply:
For older Python versions, you can include `from __future__ import annotations` and that should work.